### ISSUE TITLE
Testing drq2ppt: unkown gribcode 154 in MFP2DF

### ISSUE THREAD

**Issue by treerink** [EC-Earth/ece2cmor3#168]
I did the following test:

Generate the ppt files with the script that combines the data request (drq) and drq2ppt.py:
./generate-ec-earth-namelists.sh CMIP amip 1 1

Then I copied the generated ppt06000000 to the pptdddd0000 and ppt00000000 in the trunk/runtime/classic/ctrl directory (on KNMI's HPC facility) and tried a 2 year run.

First attempt, I get a stop (reported in the ifs.log):
`````scripts
WARNING: GRIBCODE UNKNOWN          154
UNKNOWN FIELD CODE 000154
ABOR1 CALLED
SUFPC: ABOR1 CALLED
`````
If I just take out the grib code 154 from:
`````scripts
MFP2DF = 134, 154,
`````
so replacing it by:
`````scripts
MFP2DF = 134,
`````
My short test finishes successfully.

**Comment by treerink** [comment]
I can't find grib code 154 in ifspar.json?

**Comment by klauswyser** [comment]
I suspect it's a bug in drq2ppt:
``
        if(any(mfp3dfs)):
            mfp2df.append(cmor_source.grib_code(154)) # Add surface pressure for 3D vars
``
Should it be 134? Or 152? Or remove it completely?

**Comment by etiennesky** [comment]
Hi @treerink  you have to save lnsp (152), not sp(134)! I would keep it now just in case, but as you say @klauswyser there is not real evidence we need it.

**Comment by etiennesky** [comment]
Also, it is probably another bug that sp(134) is not in ifspar.json there must be some data request for this variable

**Comment by klauswyser** [comment]
Parameter 134 is needed for some computations (RH?), see ifspar.json.

Didn't we have the discussion about which SFC pressure to save a while ago? I thought we agreed to save SFC pressure, that is 134 and only 134, or am I wrong?

@etiennesky: 134 is in the ppt file when it is produced with drq2ppt, that seems to work. The question is why there is also. Could it just be a typo in drq2ppt with the intention to always add 134?

**Comment by treerink** [comment]
@goord 154 seems a typo in drq2ppt.py. According to the Open IFS grib website: maybe check whether 129, 134, 152 have to be added instead.

**Comment by aearamos** [comment]
@treerink just a quick question, is drq2ppt generating ppt0000000000 for you? For me all I got from './generate-ec-earth-namelists.sh CMIP amip 1 1' are the two 3-hourly (pptdddddd0300) and 6-hourly (pptdddddd0600) files. Did you change the drq2ppt file?

**Comment by goord** [comment]
Yes it is a typo, it should be 152 (lnsp). As Etienne mentioned this is the preferred variable.

**Comment by klauswyser** [comment]
But 152 is not in ifspar.json while 134 is. As @etiennesky commented in #21 we will drop ln(sp) and just use sp.

**Comment by goord** [comment]
Aha point taken! I will change it to 134 then.

**Comment by goord** [comment]
BTW for the ppt00000000 and pptdddd0000 files I suggest to copy the pptdddd0600 file

**Comment by klauswyser** [comment]
> I suggest to copy the pptdddd0600 file

That works, but you won't get the orography (parameter 129) or land-sea mask (parameter 172) in the ICMGG+000000 file. Does it matter? Is it needed anywhere? These constant files are in ICMGG*INIT anyway.

**Comment by treerink** [comment]
@aearamos: see my post 1 and @goord's last post. For testing sofar I used the pptdddd0600 for both ppt00000000 and pptdddd0000. And yes indeed only pptdddd0300 and pptdddd0600 are produced by the script.

**Comment by aearamos** [comment]
Thank you, @treerink. That was my initial idea too, to just edit pptdddd0600 and use it for both  ppt00000000 and pptdddd0000. I'll have a look at the posts you mentioned.

**Comment by etiennesky** [comment]
As @klauswyser mentioned maybe we should get the orography (parameter 129) or land-sea mask (parameter 172) from the  ICMGG*INIT  files, instead of saving them in the ICMGG+000000. Using a different variable list in ppt00000000  has caused problems in the past since the first timestep has different number of variables than the other timesteps.  Maybe we can just NOT have a  ppt00000000 file, and IFS will use pptdddd0000 for the first timestep.

**Comment by etiennesky** [comment]
right! I suggest to make the runscript flexible for the ppt0000000000 file. If it is not present simply link it to the pptdddddd0600 (or pptdddddd0000) file.

Also, we must be careful about pptdddddd0300 files, we learned the hard way that if you output every 3h ALL accumulated fluxes must be in the 3h  output, or else there will be large errors.

Maybe we should move the ppt files discussion to the issue I opened last year on the portal about flexible ppt files handling.

**Comment by aearamos** [comment]
I added this to my './generate-ec-earth-namelists.sh CMIP amip 1 1' script right after drq2ppt is called:

# Since drq2ppt is not generating the ppt0000000000 file, I copied the 6-hourly file and added the two extra variables needed to start running ifs, 129 (geopotential) and 172 (land-sea mask)
 
cp pptdddddd0600 ppt0000000000
 sed -i -e 's/MFPPHY =/MFPPHY = 129, 172,/g' ppt0000000000

Then I added ppt0000000000 to be moved together with the pptd files.

**Comment by goord** [comment]
@etiennesky I do not understand your comment about the fluxes; do you imply they are accumulated over the post-processing timestep NFRPOS instead of their output timestep? This should be corrected for in ece2cmor3, because it will underestimate the accumulated quantities in such situations (which I haven't witnessed yet).

**Comment by goord** [comment]
To all: should I make drq2ppt create the initial ppt file or is this up to the user?

**Comment by klauswyser** [comment]
I would leave it to the user to create the initial file. Some users may want parameters 129 and 172 in the initial ouptut file, others don't. I can even imaging users wanting to add even more information (std deviation of orography, lakes, etc) to the initial output file, not necessarily for CMIP6 but for other projects. It's pretty simple to either link pptdddddd0600 to ppt000000000000 at runtime, or to take a copy of pptdddddd0600 and edit it, but it's hard to foresee all possible use cases. Therefore I would go for the minimal solution and let drq2ppt provide only the basic 6-hourly and 3-hourly ppt files. My 5 cents...

**Comment by aearamos** [comment]
Makes sense to me to leave it to us. The ppt0000000000 can be easily generated the way I did, but others may have different needs or may want to add different variables. It's an easy fix, so it's better to leave it to the user.

**Comment by yruprich** [comment]
@goord , about the accumulated quantities, they are indeed accumulated over the post-processeing timestep NFRPOS and reset to 0 at the beginning of each new post-processing timestep. Then, by outputing these quantities with a longer timestep, part of the integration is discarded and those outputs are then useless.

I don't think this can be corrected afterwards by ece2cmor. A control should be done before to check that the output frequency of these particular quantities is the same as NFRPOS.

PS: this is also valid for min/max quantity type.

**Comment by yruprich** [comment]
@goord , I really don't think it is a good idea to try to correct afterwards. at BSC, we tested the difference between a correct output timestep (i.e. equal to the post-processing one) and a wrong one. Even by dividing by nfrpos, we have difference reaching +/-20% of the actual values.

For example, for the solar fluxes, we miss every day the maximum of radiation over some regions of the globe. The difference between the correct and the wrong case gives patterns that look like satellite tracks with a latitudinal wave number of 8 (= 24h / 3h). This is nice but it clearly means that we cannot use such fluxes. Especially, heat budgets cannot be closed.

**Comment by klauswyser** [comment]
How many fluxes are saved every 3 hours and how many fluxes are saved every 6 hours? My impression is that *all* fluxes are saved every 3 hours, so this whole discussion is kind of superfluous and we can just abandon it, Please correct if I am wrong.

**Comment by etiennesky** [comment]
Hi @klauswyser I am afraid we cannot make the assumption that ALL fluxes are requested every 3h in the data requests. This should be checked, and depends on the data request which are not specific to IFS..

**Comment by goord** [comment]
I have a list of accumulated grib codes in ece2cmor3, I can force them to be at the highest frequency in drq2ppt, irrespective of the request frequency

**Comment by etiennesky** [comment]
that would be good, I think. 

Aren't there time-average fields such as  e.g. mx2t ? I know we won't be outputting that variable specifically (instead use mx2t24) , but there might be some other non-accumulated variables which are defined as "since previous post-processing".

Maximum temperature at 2 metres since previous post-processing | mx2t
